lib/ex_aws/s3/download.ex

```diff
@@ -86,10 +86,10 @@ defimpl ExAws.Operation, for: ExAws.S3.Download do
 
       {:ok, :done}
     rescue
-      _ ->
+      e ->
         File.close(file)
         File.rm(op.dest)
-        {:error, "error downloading file"}
+        {:error, "error downloading file", e}
```

Comment:
The problem with this change is that it changes the return type, so anyone matching against it to detect an error will suddenly have their code crashing rather than handling it. Given that the `"error downloading file"` text is pretty redundant (after all, you've tried to download and got an `:error` back, so of course that's what happened), I think I'd be happier if that string was just replaced with the error term. It's less likely that anyone is matching explicitly against that string than it is that they're matching a 2-tuple.

Reply:
@bernardd I agree, this might bite somebody. Changed to a 2-tuple.